src/transformers/generation/utils.py

```diff
@@ -1150,12 +1150,22 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
 
         # Encoder-Decoder models may also need Encoder arguments from `model_kwargs`
         if self.config.is_encoder_decoder:
+            base_model = getattr(self, self.base_model_prefix, None)
```

Comment:
to deal with `BartForConditionalGeneration`, where the `encoder` and `decoder` is inside `self.model` instead of `self`